### PR TITLE
Fixed reshape bug in `pad_shorter_ping` and added test for EK80 missing `receiver_sampling_freq`  error [all tests ci]

### PR DESCRIPTION
**Pull request by praneethratna**
Fixed reshape bug in `pad_shorter_ping` and added text case to validate the fix for #1217 which was fixed in #1219.

CC @leewujung

### CODE REVIEW

File: echopype/tests/calibrate/test_calibrate_ek80.py [hunk 1]
```diff
@@ -251,4 +251,19 @@ def test_ek80_BB_power_echoview(ek80_path):
...
     assert np.allclose(ev_vals[:, 69:8284], ep_vals[:, 69:], atol=1e-4)
-    assert np.allclose(ev_vals[:, 90:8284], ep_vals[:, 90:], atol=1e-5)
\ No newline at end of file
+    assert np.allclose(ev_vals[:, 90:8284], ep_vals[:, 90:], atol=1e-5)
+
+
+def test_ek80_CW_complex_Sv_receiver_sampling_freq(ek80_path):
+    ek80_raw_path = str(ek80_path.joinpath("D20230804-T083032.raw"))
+    ed = ep.open_raw(ek80_raw_path, sonar_model="EK80")
+
```
Comment (leewujung) [inline echopype/tests/calibrate/test_calibrate_ek80.py:260]: I suggest adding the following here to make it clear what gets changed in `compute_Sv` later.
```suggestion
# Parsed receiver_sampling_frequency is 0
assert ed["Vendor_specific"]["receiver_sampling_frequency"] == 0
```

File: echopype/tests/calibrate/test_calibrate_ek80.py [hunk 2]
```diff
@@ -251,4 +251,19 @@ def test_ek80_BB_power_echoview(ek80_path):
...
+    # Calibration object
+    waveform_mode = "CW"
+    encode_mode = "complex"
+    ds_Sv = ep.calibrate.compute_Sv(
+        ed, waveform_mode=waveform_mode, encode_mode=encode_mode
+    )
+
+    assert ds_Sv["receiver_sampling_frequency"] is not None
+    assert np.allclose(ds_Sv["receiver_sampling_frequency"].data, 1500000)
```
Comment (leewujung) [inline echopype/tests/calibrate/test_calibrate_ek80.py:269]: Suggest comments for clarity here too:
```suggestion
    # receiver_sampling_frequency substituted with default value in compute_Sv
    assert ds_Sv["receiver_sampling_frequency"] is not None
    assert np.allclose(ds_Sv["receiver_sampling_frequency"].data, 1500000)
```

**Review by leewujung — COMMENTED** [review]
@praneethratna : I pushed a change (see comment above) and made 2 small comments on the tests. Once those 2 are added I think this PR is good to go. Thanks for investigating the bug in `pad_shorter_ping` failure and adding the test!

File: echopype/convert/parse_base.py [hunk 3]
```diff
@@ -600,8 +600,11 @@ def pad_shorter_ping(data_list) -> np.ndarray:
         lens = np.array([len(item) for item in data_list])
         if np.unique(lens).size != 1:  # if some pings have different lengths along range
             if data_list[0].ndim == 2:
-                # Angle data have an extra dimension for alongship and athwartship samples
-                mask = lens[:, None, None] > np.array([np.arange(lens.max())] * 2).T
+                # Data may have an extra dimension:
+                #  - Angle data have an extra dimension for alongship and athwartship samples
+                #  - Complex data have an extra dimension for different transducer sectors
+                mask = lens[:, None, None] > np.array([np.arange(lens.max())] * data_list[0].shape[1]).T
+
```
[the remark this answers is not in the document]
Comment (praneethratna) [reply echopype/convert/parse_base.py]: understood, thanks!

**Comment by leewujung** [comment]
Hey @praneethratna : I just fixed the indent in the test that errored out. I think we're good to go now, so I'll merge this. Thanks!

**Pull request merged**